src/assertions.cr

```diff
@@ -338,7 +338,11 @@ module Minitest
 
     private def reopen(src, dst)
       backup_fd = LibC.dup(src.fd)
-      raise Errno.new("dup") if backup_fd == -1
+      {% if compare_versions(Crystal::VERSION, "0.34.0-0") > 0 %}
```

Comment:
I wish we'd have a `defined?` macro, we could have `{% if defined?(IO::Error) %}` instead of hardcoding useless version numbers.

Reply:
Actually, let's not bother with supporting Crystal 0.33 and below. Next release will be a bugfix release for Crystal 0.34 compatibility (like most previous releases have been).

Reply:
There is `{{ IO.has_constant?(:Error) }}` available but `IO::Error` already exists before 0.34.

I notice that there is no top-level version of `has_constant?` 🤔 

I can't agree with dropping the support for Crystal 0.33. The CI of shards uses latest release of Crystal. Without having been able to run specs in both Crystal version the development become bumpy.